config/routes.rb

```diff
@@ -236,6 +236,12 @@
                                           as: 'set_solution_type'
   resources :questions, only: [:edit, :update]
 
+  post 'quiz_certificates/:id/claim', to: 'quiz_certificates#claim',
+                                     as: 'claim_quiz_certificate'
+
+  get 'quiz_certificates/validate', to: 'quiz_certificates#validate',
```

Comment:
Accessing `http://localhost:3000/quiz_certificates/validate` leads to NoMethodError:
```
  def certificate_params
    params.require(:certificate).permit(:code)
  end
end
```

Reply:
I changed tthe request type to POST and unscoped the form. You can validate certificates only if you are a tutor. Then you can go to your tutorials page , e.g. `http://localhost:3000/lectures/1/tutorials` where you will find a button "Validate Quiz Certificate". Once you click that, a modal will appear where you can enter the code and validate it.